2024-07-11 12:28:07.245419: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:28:07.245610: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:28:15,784 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:28:15,886 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:28:15,887 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:28:15,887 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:28:16,977 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:28:16,977 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:16,994 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.46}
INFO flwr 2024-07-11 12:28:16,994 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:28:17,296 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:17,318 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:28:17,320 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,328 | server.py:131 | fit progress: (1, 0.41572617119374156, {'accuracy': 0.7966666666666666}, 0.333640299999999)
DEBUG flwr 2024-07-11 12:28:17,328 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:17,350 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:28:17,350 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:28:17,350 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:17,377 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,387 | server.py:131 | fit progress: (2, 0.4099652352623727, {'accuracy': 0.8}, 0.39291639999999894)
DEBUG flwr 2024-07-11 12:28:17,387 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,408 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,408 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,437 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,448 | server.py:131 | fit progress: (3, 0.4090299379927553, {'accuracy': 0.8066666666666666}, 0.4532778000000004)
DEBUG flwr 2024-07-11 12:28:17,448 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,469 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,469 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,494 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,505 | server.py:131 | fit progress: (4, 0.4090304235570652, {'accuracy': 0.8066666666666666}, 0.5099806999999998)
DEBUG flwr 2024-07-11 12:28:17,505 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,525 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,525 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:17,555 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,567 | server.py:131 | fit progress: (5, 0.4090303169882947, {'accuracy': 0.8066666666666666}, 0.5733820000000005)
DEBUG flwr 2024-07-11 12:28:17,568 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:17,592 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,592 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:17,627 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,639 | server.py:131 | fit progress: (6, 0.41041855575204483, {'accuracy': 0.8066666666666666}, 0.6443402999999996)
DEBUG flwr 2024-07-11 12:28:17,639 | server.py:178 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,667 | server.py:192 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,667 | server.py:227 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,703 | server.py:241 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,715 | server.py:131 | fit progress: (7, 0.4101312315189953, {'accuracy': 0.81}, 0.7207507)
DEBUG flwr 2024-07-11 12:28:17,716 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,743 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,744 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,784 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,797 | server.py:131 | fit progress: (8, 0.4101307082509111, {'accuracy': 0.81}, 0.8028507000000005)
DEBUG flwr 2024-07-11 12:28:17,797 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,833 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,833 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,878 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,890 | server.py:131 | fit progress: (9, 0.41013069859464707, {'accuracy': 0.81}, 0.8962237000000002)
DEBUG flwr 2024-07-11 12:28:17,891 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,919 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:17,920 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:17,962 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:17,975 | server.py:131 | fit progress: (10, 0.4101306986284224, {'accuracy': 0.81}, 0.9807764999999993)
DEBUG flwr 2024-07-11 12:28:17,976 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:18,007 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:28:18,008 | server.py:153 | FL finished in 1.0129579999999994
INFO flwr 2024-07-11 12:28:18,008 | app.py:225 | app_fit: losses_distributed [(1, 0.4105173237621784), (2, 0.4093971143166224), (3, 0.40935479700565336), (4, 0.40935504833857217), (5, 0.4155810562701061), (6, 0.4096508579594748), (7, 0.4094183376857213), (8, 0.4094184249639511), (9, 0.4094184249639511), (10, 0.4094184249639511)]
INFO flwr 2024-07-11 12:28:18,008 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:28:18,008 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:28:18,008 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.41572617119374156), (2, 0.4099652352623727), (3, 0.4090299379927553), (4, 0.4090304235570652), (5, 0.4090303169882947), (6, 0.41041855575204483), (7, 0.4101312315189953), (8, 0.4101307082509111), (9, 0.41013069859464707), (10, 0.4101306986284224)]
INFO flwr 2024-07-11 12:28:18,009 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.46), (1, 0.7966666666666666), (2, 0.8), (3, 0.8066666666666666), (4, 0.8066666666666666), (5, 0.8066666666666666), (6, 0.8066666666666666), (7, 0.81), (8, 0.81), (9, 0.81), (10, 0.81)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95290736,  1.05748238,  0.91940811,  0.42151183,  0.21126978,
        -0.00611663, -0.10322159, -0.07551377]]), array([0.07578311])]
[array([[-1.95290736,  1.05748238,  0.91940811,  0.42151183,  0.21126978,
        -0.00611663, -0.10322159, -0.07551377]]), array([0.07578311])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97226857,  1.02033874,  0.80172342,  0.43390302,  0.18081885,
         0.02353493, -0.05501256, -0.06904044]]), array([0.17879413])]
[array([[-1.97226857,  1.02033874,  0.80172342,  0.43390302,  0.18081885,
         0.02353493, -0.05501256, -0.06904044]]), array([0.17879413])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95337921,  1.02891065,  0.79189575,  0.43669926,  0.17556429,
         0.03246815, -0.02588775, -0.06589957]]), array([0.11343185])]
[array([[-1.95337921,  1.02891065,  0.79189575,  0.43669926,  0.17556429,
         0.03246815, -0.02588775, -0.06589957]]), array([0.11343185])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95329284,  1.0289178 ,  0.79193608,  0.43667998,  0.17561253,
         0.03246408, -0.02582221, -0.06599923]]), array([0.1132583])]
[array([[-1.95329284,  1.0289178 ,  0.79193608,  0.43667998,  0.17561253,
         0.03246408, -0.02582221, -0.06599923]]), array([0.1132583])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95329832,  1.0289198 ,  0.79193641,  0.4366822 ,  0.1756113 ,
         0.03246767, -0.02582258, -0.06600043]]), array([0.11325969])]
[array([[-1.95329832,  1.0289198 ,  0.79193641,  0.4366822 ,  0.1756113 ,
         0.03246767, -0.02582258, -0.06600043]]), array([0.11325969])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.91451829,  0.97981543,  0.80980209,  0.37785345,  0.17348693,
         0.04868374, -0.04049101, -0.04764563]]), array([0.15858998])]
[array([[-1.91451829,  0.97981543,  0.80980209,  0.37785345,  0.17348693,
         0.04868374, -0.04049101, -0.04764563]]), array([0.15858998])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96577461,  1.00070323,  0.83505207,  0.37470639,  0.15906121,
         0.0774635 , -0.02728782, -0.05172874]]), array([0.16868551])]
[array([[-1.96577461,  1.00070323,  0.83505207,  0.37470639,  0.15906121,
         0.0774635 , -0.02728782, -0.05172874]]), array([0.16868551])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570247,  1.00067134,  0.83499901,  0.37471524,  0.15903259,
         0.07751972, -0.02729271, -0.05174014]]), array([0.1686726])]
[array([[-1.96570247,  1.00067134,  0.83499901,  0.37471524,  0.15903259,
         0.07751972, -0.02729271, -0.05174014]]), array([0.1686726])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867244])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867244])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
>>>>>>>>>>>>>run time:
0.546875s
